src/main/java/org/folio/edge/sip2/repositories/PatronRepository.java

```diff
@@ -0,0 +1,354 @@
+      }
+    } else {
+      overdueItemsCount = 0;
+      overdueItems = null;
+    }
+
+    return builder.overdueItemsCount(Integer.valueOf(overdueItemsCount)).overdueItems(overdueItems);
+  }
+
+  @SuppressWarnings("rawtypes")
+  private Future<PatronInformationResponseBuilder> addRecalls(List<Future<JsonObject>> recalls,
+      Integer startItem, Integer endItem, boolean details,
+      PatronInformationResponseBuilder builder) {
+    return CompositeFuture.all(new ArrayList<Future>(recalls)).map(result -> {
+      final int recallItemsCount = Math.min(countRecallItems(recalls), 9999);
+      final List<String> recallItems;
+
+      if (details) {
+        recallItems = getRecallItems(recalls, startItem, endItem);
+      } else {
+        recallItems = null;
+      }
+
+      return builder.recallItemsCount(Integer.valueOf(recallItemsCount)).recallItems(recallItems);
+    });
+  }
+
+  private String getPatronPersonalName(JsonObject user) {
+    final JsonObject personal = user.getJsonObject(FIELD_PERSONAL, new JsonObject());
+    final Optional<String> firstName = Optional.ofNullable(personal.getString("firstName"));
+    final Optional<String> middleName = Optional.ofNullable(personal.getString("middleName"));
+    final Optional<String> lastName = Optional.ofNullable(personal.getString("lastName"));
+
+    return Stream.of(firstName, middleName, lastName)
+        .filter(Optional::isPresent)
+        .map(Optional::get)
+        .collect(Collectors.joining(" "));
+  }
+
+  private String getPatronHomeAddress(JsonObject user) {
+    final JsonObject personal = user.getJsonObject(FIELD_PERSONAL, new JsonObject());
+    final JsonArray addresses = personal.getJsonArray("addresses", new JsonArray());
+    // For now, we will do the following:
+    // 1. if the address list > 0, pick the "primaryAddress"
+    // 2. if no "primaryAddress", pick the first address
+    // 3. no addresses, return null (home address is an optional field)
+    // In the future, we should use the UUID of the home address type to select the home address
+    // and if no match, then return null so as to not expose other addresses.
+    // This can be cleaned up a bit when we get to Java 11.
+    Optional<String> addressString = addresses.stream()
+        .map(o -> (JsonObject) o)
+        .filter(address -> address.getBoolean("primaryAddress", FALSE).booleanValue())
+        .findFirst()
+        .map(Optional::of)
+        .orElseGet(() -> addresses.stream()
+            .map(o -> (JsonObject) o)
+            .findFirst()
+            .map(Optional::of)
+            .orElse(Optional.empty()))
+        .map(this::toHomeAddress);
+
+    return addressString.isPresent() ? addressString.get() : null;
+  }
+
+  private String getPatronEmailAddress(JsonObject user) {
+    final JsonObject personal = user.getJsonObject(FIELD_PERSONAL, new JsonObject());
+
+    return personal.getString("email");
+  }
+
+  private String getPatronHomePhoneNumber(JsonObject user) {
```

Comment:
We could put them somewhere else, though I don't think there would be any other consumer of this logic. I'm not keen on exposing implementation to other classes. Dependencies may be formed by mistake or intentionally. I'll think about this as well.

Reply:
Maybe we could add a JSON utility class to help with common tasks. I am noticing I am doing a lot of the same things, like:
`jsonObject.getJsonObject("some-object", new JsonObject()).getString("some-field")`
Which basically means that I want to get a field within a child object, but if the object is not present, I don't want a NPE, so build a new `JsonObject` before calling `getString`. Since the object should be eagerly created as a method argument, it would probably be better to check for `null` first. That would make the code look like blah, so I'll just move these types of operations to a JSON utility class.